src/js/ripple/seed.js

```diff
@@ -57,104 +62,105 @@ Seed.prototype.parse_passphrase = function(j) {
   return this;
 };
 
+Seed.prototype.parse_base58 = function(j) {
+  var result = Base.decode_multi(j, Seed.width);
+
+  if (result.error) {
+    this._value = NaN;
+  } else if (lodash.isEqual(result.version, [Base.VER_FAMILY_SEED]) ||
+        lodash.isEqual(result.version, Base.VER_ED25519_SEED)) {
+      this._version = result.version;
+      this._value = sjcl.bn.fromBits(sjcl.codec.bytes.toBits(result.bytes));
+  }
+
+  return this;
+};
+
 Seed.prototype.to_json = function() {
   if (!(this.is_valid())) {
     return NaN;
   }
 
-  var output = Base.encode_check(Base.VER_FAMILY_SEED, this.to_bytes());
+  var version = this._version; // Base.VER_FAMILY_SEED;
+  var output = Base.encode_check(version, this.to_bytes());
 
   return output;
 };
 
-function append_int(a, i) {
-  return [].concat(a, i >> 24, (i >> 16) & 0xff, (i >> 8) & 0xff, i & 0xff);
-}
-
-function firstHalfOfSHA512(bytes) {
-  return sjcl.bitArray.bitSlice(
-    sjcl.hash.sha512.hash(sjcl.codec.bytes.toBits(bytes)),
-    0, 256
-  );
-}
-
-// Removed a `*` so this JSDoc-ish syntax is ignored.
-// This will soon all change anyway.
-/*
-* @param account
-*        {undefined}                 take first, default, KeyPair
-*
-*        {Number}                    specifies the account number of the KeyPair
-*                                    desired.
-*
-*        {Uint160} (from_json able), specifies the address matching the KeyPair
-*                                    that is desired.
+Seed.prototype.get_key = function(opts) {
```

Comment:
We might also want to deprecate this so we can phase out this API, so eventually the only way to get a keypair is through the keypairs service. That would mean removing the keypair.js file, and updating the Secp256k1Pair to no longer delegate.

Given we're breaking it now, throwing an error unless the argument is an object, you could argue that we may as well go ahead now and remove the api entirely.